src/discussions/comments/data/thunks.js

```diff
@@ -120,3 +128,16 @@ export function removeComment(commentId, threadId) {
     }
   };
 }
+
+export function fetchCourseSettings(courseId) {
+  return async (dispatch) => {
+    try {
+      dispatch(fetchCourseSettingsRequest({ courseId }));
+      const data = await getCourseSettings(courseId);
+      dispatch(fetchCourseSettingsSuccess(camelCaseObject(data)));
+    } catch (error) {
+      dispatch(fetchCourseSettingsFailed());
+      logError(error);
```

Comment:
Regular users will get a permission error here. So a permission denied isn't an error here.

Reply:
> Regular users will get a permission error here. So a permission denied isn't an error here.

@xitij2000 
So, what do you suggest? The dispatch only sets the state but doesn't raise any error. Should we raise errors instead?

Reply:
Check if it's a permission error and in that case, don't mark it as a failed request.